biosppy/signals/ecg.py

```diff
@@ -1381,6 +1381,858 @@ def ASI_segmenter(signal=None, sampling_rate=1000.0, Pth=5.0):
+
+    Parameters
+    ----------
+    
+    ecg : array
+        Input raw ECG signal.
+    fs: int, float, optional
+        Sampling frequency (Hz).
+
+    Returns
+    ----------
+    qrs_i_raw: array
+        R-peak location indices.
+
+    References
+    ----------
+    .. [MdNiNai22] Khan, Naimul and Imtiaz, Md Niaz, "Pan-Tompkins++: A Robust Approach to Detect R-peaks in ECG Signals", 
+        arXiv preprint arXiv:2211.03171, 2022
+
+    """
+
+    # check inputs
+    if ecg is None:
+        raise TypeError("Please specify an input signal.")
+        
+    ''' Initialize '''
+    delay = 0
+    skip = 0                    # Becomes one when a T wave is detected
+    m_selected_RR = 0
+    mean_RR = 0
+    ser_back = 0
+
+
+    ''' Noise Cancelation (Filtering) (5-18 Hz) '''
+
+    if fs == 200:
+        ''' Remove the mean of Signal '''
+        #If fs=200 keep frwquency 5-12 Hz otherwise 5-18 Hz
+        ecg = ecg - np.mean(ecg)  
+
+        Wn = 12*2/fs
+        N = 3
+        a, b = signal.butter(N, Wn, btype='lowpass')
+        ecg_l = signal.filtfilt(a, b, ecg)
+        
+        ecg_l = ecg_l/np.max(np.abs(ecg_l)) #Normalize by dividing high value. That reduce time of calculation
+
+        Wn = 5*2/fs
+        N = 3                                           # Order of 3 less processing
+        a, b = signal.butter(N, Wn, btype='highpass')             # Bandpass filtering
+        ecg_h = signal.filtfilt(a, b, ecg_l, padlen=3*(max(len(a), len(b))-1))
+        ecg_h = ecg_h/np.max(np.abs(ecg_h))  #Normalize by dividing high value. That reduce time of calculation
+
+    else:
+        ''' Band Pass Filter for noise cancelation of other sampling frequencies (Filtering)'''
+        f1 = 5 #3 #5                                          # cutoff low frequency to get rid of baseline wander
+        f2 = 18 #25  #15                                         # cutoff frequency to discard high frequency noise
+        Wn = [f1*2/fs, f2*2/fs]                         # cutoff based on fs
+        N = 3                                           
+        a, b = signal.butter(N=N, Wn=Wn, btype='bandpass')   # Bandpass filtering
+        ecg_h = signal.filtfilt(a, b, ecg, padlen=3*(max(len(a), len(b)) - 1))
+                    
+        ecg_h = ecg_h/np.max(np.abs(ecg_h))
+
+    vector = [1, 2, 0, -2, -1]
+    if fs != 200:
+        int_c = 160/fs
+        b = interp1d(range(1, 6), [i*fs/8 for i in vector])(np.arange(1, 5.1, int_c))  
+                                                                                    
+    else:
+        b = [i*fs/8 for i in vector]      
+
+    ecg_d = signal.filtfilt(b, 1, ecg_h, padlen=3*(max(len(a), len(b)) - 1))
+
+    ecg_d = ecg_d/np.max(ecg_d)
+
+
+    ''' Squaring nonlinearly enhance the dominant peaks '''
+
+    ecg_s = ecg_d**2
+    
+    #Smooting
+    sm_size = int(0.06 * fs)       
+    ecg_s = smoother(signal=ecg_s, kernel='flattop', size=sm_size, mirror=True)
```

Comment:
This is the biosppy function from biosppy.signals.tools, right? It is imported as "st"

```suggestion
    ecg_s = st.smoother(signal=ecg_s, kernel='flattop', size=sm_size, mirror=True)
```